public/app/features/dashboard-scene/sharing/components/ImagePreview.tsx

```diff
@@ -0,0 +1,119 @@
+import { css } from '@emotion/css';
+import { useMemo, useEffect } from 'react';
+import { useMeasure } from 'react-use';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { t } from '@grafana/i18n';
+import { Alert, LoadingBar, Text, useStyles2 } from '@grafana/ui';
+
+type ErrorState = {
+  message: string;
+  title: string;
+} | null;
+
+interface ImagePreviewProps {
+  imageBlob: Blob | null;
+  isLoading: boolean;
+  error: ErrorState;
+  title?: string;
```

Comment:
this shouldn't be optional because there's always a title

Reply:
this title is used as part of loading skeleton when image is loading in and is a "nice to have" / optional in case we want to use this in other places and not worry about showing any text